paasta_tools/cli/cmds/spark_run.py

```diff
@@ -606,6 +650,53 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     return spark_app_name
 
 
+def _calculate_docker_memory_limit(
+    spark_conf: Mapping[str, str], memory_limit: str
+) -> str:
+    """ In Order of preference:
+    1. Argument: --docker-memory-limit
+    2. --spark-args or spark-submit: spark.driver.memory
+    3. Default
+    """
+    try:
+        if memory_limit:
+            docker_memory_limit_str = memory_limit
+            adjustment_factor = 1
+        else:
+            docker_memory_limit_str = spark_conf.get(
+                "spark.driver.memory", DEFAULT_DRIVER_MEMORY_BY_SPARK
+            )
+            adjustment_factor = DOCKER_RESOURCE_ADJUSTMENT_FACTOR
+        match = re.match(r"([0-9]+)([a-z]*)", docker_memory_limit_str)
+        memory_val = int(match[1]) * adjustment_factor
+        memory_unit = match[2]
+        docker_memory_limit = f"{memory_val}{memory_unit}"
+    except Exception as e:
+        # For any reason it fails, continue with default value
+        log.error(
+            f"Failed to parse docker memory limit. Error: {e}. Example values: 1g, 200m."
+            " Setting a default value {DEFAULT_MAX_DOCKER_MEMORY_LIMIT}"
+        )
+        docker_memory_limit = DEFAULT_MAX_DOCKER_MEMORY_LIMIT
+
+    return docker_memory_limit
+
+
+def _calculate_docker_cpu_limit(spark_conf: Mapping[str, str], cpu_limit: str) -> str:
+    """ In Order of preference:
+    1. Argument: --docker-cpu-limit
+    2. --spark-args or spark-submit: spark.driver.cores
+    3. Default
+    """
+    if cpu_limit:
+        docker_cpu_limit = cpu_limit
+    else:
+        docker_cpu_limit = str(
+            int(spark_conf.get("spark.driver.cores", DEFAULT_DRIVER_CORES_BY_SPARK))
+        )
+    return docker_cpu_limit
```

Comment:
feel free to drop this suggestion:
```suggestion
    return cpu_limit if cpu_limit else spark_conf.get("spark.driver.cores", str(DEFAULT_DRIVER_CORES_BY_SPARK))
```

(from my understanding `spark.driver.cores` will always be a string, so we really just want to cast the default core value)

Reply:
Added back.